### ISSUE TITLE
Error in `.publish` after `.remove` and `.patch` is not caught and exposes user data

### ISSUE THREAD

**Issue by panstromek** [feathersjs/feathers#1707]
I have this code:
```js
app.service(service).publish(entity => {
   return app.channel(`sessions/${entity.session_id}/admin`)
})
```
In my case, `entity` was undefined, function threw an exception and feathers dumped bunch of data back to the client, including user with hashed password, socketId and other internal data. It also ignored global error hooks completely. 

It also seems happenning only with `patch` and `remove` method. Doesn't happen with `.create`, that one passed through error hooks (which sanitizes the data).

Node: 12.0
Feathers: 4.3.11

**Comment by panstromek** [comment]
If you have problems reproducing, I'll put together a repo or smth, I don't have much time at the moment ;)

**Comment by daffl** [comment]
How did `entity` end up being `undefined`? A service method should always return an object, otherwise weird things will happen indeed.

**Comment by panstromek** [comment]
That was a bug on my part, but that's not really relevant, that's just how I found it. You can trigger the error in any different way in that function (like `(undefined).not_here` or something).

**Comment by daffl** [comment]
Would it be possible to provide a complete simplified example how this could be happening?

**Comment by panstromek** [comment]
Ok, I created a repo.
https://github.com/panstromek/feathers-bug-report

It's just vanilla feathers app with an error thrown in `publish` method. Then there is a report script that just tries to create message and recieves this error that doesn't go through global error hooks.

```js
 app.publish((data, hook) => {
    console.log('Publishing an event - now I crash') 
    throw new Unprocessable()
    return app.channel('authenticated')
  })
```
and
```js
feathers.service('messages').create({}).then(data => {
  //
}).catch(data => {

  console.log(data)
  console.log('Look at the error above, it\'s not usual Feathers error and contains a lot of internal information.')
})
```
I hope that'll do 😉

**Comment by daffl** [comment]
Errors in publishers should indeed be handled more gracefully but I am not able to reproduce that any sensitive information gets propagated to a client. Sure you will see it on the server but it's up to you if you want to log it all or not.

**Comment by panstromek** [comment]
Well, the error that gets logged is logged from client - I created a client in that `report.js` file, it's not a server log. It contains the whole params object (which usually contains user object with all fields. I ommited the auth in the example, though).

**Comment by panstromek** [comment]
I updated the repo with a hook
```js
  before: {
    all: [ctx => {
      ctx.params.user = {
        password: 'this should be a secret'
      }
    }],
```
and this gets logged into console from the client.
![image](https://user-images.githubusercontent.com/25121817/69760288-69580780-1164-11ea-9e6b-c3e07f672d3e.png)

**Comment by daffl** [comment]
Interesting. I didn't see it in the chat but might be happening because event dispatching happens synchronously. Either way, this has been fixed in v4.4.1, thank you for reporting and digging into the issue.

**Comment by panstromek** [comment]
Cool, thanks 😉
